models/classes/resources/relation/service/ResourceRelationServiceProxy.php

```diff
@@ -80,7 +87,31 @@ public function findRelations(FindAllQuery $query): ResourceRelationCollection
 
     private function getResourceRelationService(string $serviceId): ResourceRelationServiceInterface
     {
-        return $this->getServiceLocator()->get($serviceId);
+        try {
+            return $this->getServiceManager()->get($serviceId);
+        } catch (ServiceNotFoundException $exception) {
+            return $this->getServiceManager()->getContainer()->get($serviceId);
+        }
```

Comment:
No need for it, the container does it automatically for you :)

```suggestion
        return $this->getServiceManager()->getContainer()->get($serviceId);
```